### DISCUSSION TITLE
Vits optimizer and scheduler

### DISCUSSION THREAD

**Discussion by amitaie** [coqui-ai/TTS#2176] [General Q&A]
Hey, 
I think that the schedulers of Vits in the wrong order:

> def get_optimizer(self) -> List:
>         **optimizer0** = get_optimizer(self.config.optimizer, self.config.optimizer_params, self.config.lr_disc, **self.disc**)
>         gen_parameters = chain(params for k, params in self.named_parameters() if not k.startswith("disc."))
>         optimizer1 = get_optimizer(
>             self.config.optimizer, self.config.optimizer_params, self.config.lr_gen, parameters=gen_parameters)
>         return [**optimizer0**, optimizer1]

on the other hand:

>  def get_scheduler(self, optimizer) -> List:
>         **scheduler_G** = get_scheduler(self.config.lr_scheduler_gen, self.config.lr_scheduler_gen_params, **optimizer[0]**)
>         scheduler_D = get_scheduler(self.config.lr_scheduler_disc, self.config.lr_scheduler_disc_params, optimizer[1])
>         return [scheduler_D, scheduler_G]

Am i right? I'm not sure how does it effect the training.

**Comment by erogol** [comment]
yeah its a bug but since we use the same values I don't think it affects our training recipes.

**Comment by erogol** [comment]
Fixed it on dev. Thanks!
